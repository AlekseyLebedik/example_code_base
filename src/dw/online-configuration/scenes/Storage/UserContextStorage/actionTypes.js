export const USER_CONTEXT_STORAGE_RELOAD_DETAILS =
  'storage/USER_CONTEXT_STORAGE_RELOAD_DETAILS';
export const USER_CONTEXT_STORAGE_FETCH = 'storage/USER_CONTEXT_STORAGE_FETCH';
export const USER_CONTEXT_STORAGE_FETCH_SUCCESS =
  'storage/USER_CONTEXT_STORAGE_FETCH_SUCCESS';

export const USER_CONTEXT_STORAGE_LIST_ITEM_ONCLICK =
  'storage/USER_CONTEXT_STORAGE_LIST_ITEM_ONCLICK';

export const USER_CONTEXT_STORAGE_FILES_UPLOAD_CLOSE_MODAL =
  'storage/USER_CONTEXT_STORAGE_FILES_UPLOAD_CLOSE_MODAL';
export const USER_CONTEXT_STORAGE_FILES_UPLOAD_OPEN_MODAL =
  'storage/USER_CONTEXT_STORAGE_FILES_UPLOAD_OPEN_MODAL';
export const USER_CONTEXT_STORAGE_FILES_UPLOAD =
  'storage/USER_CONTEXT_STORAGE_FILES_UPLOAD';
export const USER_CONTEXT_STORAGE_FILES_UPLOAD_SUCCESS =
  'storage/USER_CONTEXT_STORAGE_FILES_UPLOAD_SUCCESS';

export const USER_CONTEXT_STORAGE_GET_STORAGE_CONTEXTS =
  'storage/USER_CONTEXT_STORAGE_GET_STORAGE_CONTEXTS';
export const USER_CONTEXT_STORAGE_GET_STORAGE_CONTEXTS_SUCCESS =
  'storage/USER_CONTEXT_STORAGE_GET_STORAGE_CONTEXTS_SUCCESS';
