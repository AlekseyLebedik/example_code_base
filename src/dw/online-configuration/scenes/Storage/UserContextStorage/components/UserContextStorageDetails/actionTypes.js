export const USER_CONTEXT_STORAGE_DETAILS_DELETE_FILE =
  'storage/USER_CONTEXT_STORAGE_DETAILS_DELETE_FILE';
export const USER_CONTEXT_STORAGE_DETAILS_DELETE_FILE_SUCCESS =
  'storage/USER_CONTEXT_STORAGE_DETAILS_DELETE_FILE_SUCCESS';

export const USER_CONTEXT_STORAGE_DETAILS_FILE_DOWNLOAD =
  'storage/USER_CONTEXT_STORAGE_DETAILS_FILE_DOWNLOAD';
