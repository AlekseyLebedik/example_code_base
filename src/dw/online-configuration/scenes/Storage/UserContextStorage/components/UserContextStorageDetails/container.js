import { connect } from 'dw/core/helpers/component';

import {
  deleteUserContextStorageFile,
  downloadUserContextStorageFile,
} from './actions';
import StatelessComponent from './presentational';

const stateToProps = (state, props) => ({
  ...props,
});

const dispatchToProps = dispatch => ({
  deleteFileHandler: params => dispatch(deleteUserContextStorageFile(params)),
  downloadFileHandler: params =>
    dispatch(downloadUserContextStorageFile(params)),
});

export default connect(stateToProps, dispatchToProps, StatelessComponent);
