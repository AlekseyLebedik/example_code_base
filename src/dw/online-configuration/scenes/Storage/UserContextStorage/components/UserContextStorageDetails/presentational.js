import React from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';

import Icon from '@material-ui/core/Icon';
import IconButton from '@material-ui/core/IconButton';
import Tooltip from '@material-ui/core/Tooltip';
import SectionTitle from 'dw/core/components/SectionTitle';
import ConfirmActionComponent from 'dw/core/components/ConfirmActionComponent';
import UserLink from 'dw/online-configuration/components/UserLink';
import KeyValue from 'dw/core/components/KeyValue';
import { STORAGE_DELETE_USER_CONTEXT_FILES } from '@demonware/devzone-core/access/PermissionCheck/permissions';
import { CheckPermission } from '@demonware/devzone-core/access/CheckPermissions';
import { currentEnvDetailsSelector } from 'dw/core/helpers/title-env-selectors';

import UserContextStorageDetailsEmpty from '../UserContextStorageDetailsEmpty';

const UserContextStorageDetails = props => {
  const currentEnv = useSelector(currentEnvDetailsSelector);
  const {
    selectedFile,
    deleteFileHandler,
    downloadFileHandler,
    elementsOrder,
  } = props;
  const newElementsOrder = elementsOrder.map(column => {
    switch (column) {
      case 'userID':
        return {
          name: 'userID',
          key: column,
          formatter: userId => <UserLink userId={userId} />,
        };
      default:
        return column;
    }
  });

  const Actions = () => (
    <>
      <Tooltip title="Download">
        <IconButton
          color="inherit"
          onClick={() =>
            downloadFileHandler({
              fileID: selectedFile.fileID,
              fileName: selectedFile.fileName,
              userID: selectedFile.userID,
              context: selectedFile.context,
              accountType: selectedFile.accountType,
            })
          }
        >
          <Icon>file_download</Icon>
        </IconButton>
      </Tooltip>
      <CheckPermission
        predicate={STORAGE_DELETE_USER_CONTEXT_FILES}
        object={`titleenv.${currentEnv.id}`}
      >
        <ConfirmActionComponent
          component="IconButton"
          container="details"
          tooltip="Delete"
          onClick={() =>
            deleteFileHandler({
              fileID: selectedFile.fileID,
              fileName: selectedFile.fileName,
              userID: selectedFile.userID,
              context: selectedFile.context,
              accountType: selectedFile.accountType,
            })
          }
          confirm={{
            confirmMsg: 'Are you sure you want to delete this file?',
            title: 'Confirm Delete',
            mainButtonLabel: 'Delete',
            destructive: true,
          }}
          color="inherit"
        >
          delete
        </ConfirmActionComponent>
      </CheckPermission>
    </>
  );
  const Maincomponent = () => (
    <div className="details__container flex-rows-container">
      <SectionTitle title={`${selectedFile.fileName}`}>
        {Actions()}
      </SectionTitle>
      <div className="scrollable-content">
        <KeyValue
          item={selectedFile}
          elementsOrder={newElementsOrder}
          size={4}
        />
      </div>
    </div>
  );

  return !selectedFile ? <UserContextStorageDetailsEmpty /> : Maincomponent();
};
UserContextStorageDetails.propTypes = {
  selectedFile: PropTypes.object,
  deleteFileHandler: PropTypes.func.isRequired,
  downloadFileHandler: PropTypes.func.isRequired,
  elementsOrder: PropTypes.arrayOf(PropTypes.string),
};
UserContextStorageDetails.defaultProps = {
  selectedFile: undefined,
  elementsOrder: [],
};

export default UserContextStorageDetails;
