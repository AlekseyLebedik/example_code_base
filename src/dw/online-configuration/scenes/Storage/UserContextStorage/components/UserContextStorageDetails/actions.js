import download from 'downloadjs';
import { nonCriticalHTTPError } from '@demonware/devzone-core/helpers/errors';
import * as AT from './actionTypes';
import { reloadDetails } from '../../actions';

export function deleteUserContextStorageFile(params) {
  return dispatch => {
    dispatch({
      type: AT.USER_CONTEXT_STORAGE_DETAILS_DELETE_FILE,
      params,
    });
  };
}

export function deleteUserContextStorageFileSuccess(payload) {
  return dispatch => {
    const { userID, context, accountType, fileID } = payload;
    const path = `/${userID}/${context}/${accountType}`;
    dispatch(reloadDetails(path));
    dispatch({
      type: AT.USER_CONTEXT_STORAGE_DETAILS_DELETE_FILE_SUCCESS,
      fileID,
    });
  };
}

export function deleteUserContextStorageFileFailed(err) {
  return dispatch => {
    dispatch(nonCriticalHTTPError(err));
  };
}

export function downloadUserContextStorageFile(params) {
  return dispatch => {
    dispatch({
      type: AT.USER_CONTEXT_STORAGE_DETAILS_FILE_DOWNLOAD,
      params,
    });
  };
}

export function downloadUserContextStorageFileSuccess(data) {
  return () => download(data.fileData, data.fileName);
}

export function downloadUserContextStorageFileFailed(err) {
  return dispatch => {
    dispatch(nonCriticalHTTPError(err));
  };
}
