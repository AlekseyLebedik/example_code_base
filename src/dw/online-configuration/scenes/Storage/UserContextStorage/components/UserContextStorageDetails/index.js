import UserContextStorageDetails from './container';

export default UserContextStorageDetails;
