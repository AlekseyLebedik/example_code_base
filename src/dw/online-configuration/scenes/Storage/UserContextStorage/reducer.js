import * as AT from './actionTypes';
import { USER_CONTEXT_STORAGE_DETAILS_DELETE_FILE_SUCCESS } from './components/UserContextStorageDetails/actionTypes';

const INITIAL_STATE = {
  entries: [],
  selectedFile: undefined,
  elementsOrder: [],
  q: undefined,
  contextsList: [],
  uploadModalVisible: false,
  reloadDetailPath: null,
};

export const reducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case AT.USER_CONTEXT_STORAGE_RELOAD_DETAILS:
      return {
        ...state,
        reloadDetailPath: action.path,
      };
    case AT.USER_CONTEXT_STORAGE_FETCH_SUCCESS:
      return {
        ...state,
        entries: action.append
          ? [...state.entries, ...action.entries]
          : action.entries,
        elementsOrder: action.elementsOrder,
        selectedFile: undefined,
        q: action.q,
      };
    case AT.USER_CONTEXT_STORAGE_LIST_ITEM_ONCLICK:
      return {
        ...state,
        selectedFile: action.file,
      };
    case AT.USER_CONTEXT_STORAGE_FILES_UPLOAD_CLOSE_MODAL:
      return {
        ...state,
        uploadModalVisible: false,
      };
    case AT.USER_CONTEXT_STORAGE_FILES_UPLOAD_OPEN_MODAL:
      return {
        ...state,
        uploadModalVisible: true,
      };
    case AT.USER_CONTEXT_STORAGE_FILES_UPLOAD_SUCCESS:
      return {
        ...state,
        selectedFile: action.selectedFile,
        uploadModalVisible: false,
        q: action.q,
      };
    case USER_CONTEXT_STORAGE_DETAILS_DELETE_FILE_SUCCESS:
      return {
        ...state,
        entries: state.entries.filter(file => file.fileID !== action.fileID),
        selectedFile: undefined,
      };
    case AT.USER_CONTEXT_STORAGE_GET_STORAGE_CONTEXTS_SUCCESS:
      return {
        ...state,
        contextsList: action.contextsList,
      };
    default:
      return state;
  }
};
