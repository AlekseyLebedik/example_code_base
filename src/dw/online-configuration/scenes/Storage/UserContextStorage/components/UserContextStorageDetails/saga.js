import { call, put, takeLatest } from 'redux-saga/effects';

import {
  deleteUserContextStorageFile as apiDeleteUserContextStorageFile,
  downloadUserContextStorageFile as apiDownloadContextStorageStorageFile,
} from 'dw/online-configuration/services/storages';

import * as Actions from './actions';
import * as AT from './actionTypes';

function* deleteUserContextStorageFile(action) {
  try {
    yield call(apiDeleteUserContextStorageFile, action.params);
    yield put(Actions.deleteUserContextStorageFileSuccess(action.params));
  } catch (e) {
    yield put(Actions.deleteUserContextStorageFileFailed(e));
  }
}

function* downloadUserContextStorageFile(action) {
  try {
    const { data } = yield call(
      apiDownloadContextStorageStorageFile,
      action.params
    );
    yield put(Actions.downloadUserContextStorageFileSuccess(data));
  } catch (e) {
    yield put(Actions.downloadUserContextStorageFileFailed(e));
  }
}

function* saga() {
  yield takeLatest(
    AT.USER_CONTEXT_STORAGE_DETAILS_DELETE_FILE,
    deleteUserContextStorageFile
  );
  yield takeLatest(
    AT.USER_CONTEXT_STORAGE_DETAILS_FILE_DOWNLOAD,
    downloadUserContextStorageFile
  );
}

export default saga;
