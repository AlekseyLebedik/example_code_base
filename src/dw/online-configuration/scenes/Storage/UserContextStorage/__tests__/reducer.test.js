import { USER_CONTEXT_STORAGE_DETAILS_DELETE_FILE_SUCCESS } from '../components/UserContextStorageDetails/actionTypes';
import { reducer } from '../reducer';
import * as AT from '../actionTypes';

describe('UserContextStorage', () => {
  describe('Reducer', () => {
    const initialState = reducer(undefined, {});
    const populatedState = {
      ...initialState,
      nextPageToken: 'ABC',
      elementsOrder: ['fileID'],
      uploadModalVisible: false,
      reloadDetailPath: '/a',
      entries: [{ fileID: 1 }, { fileID: 2 }],
      selectedFile: { fileID: 1 },
      contextsList: ['x360', 'x1'],
    };

    it('returns default state', () => {
      const state = reducer(undefined, {});
      expect(state).toMatchSnapshot();
    });

    describe('USER_CONTEXT_STORAGE_RELOAD_DETAILS', () => {
      it('sets the reload details path', () => {
        const action = {
          type: AT.USER_CONTEXT_STORAGE_RELOAD_DETAILS,
          path: '/b',
        };
        const state = reducer(populatedState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('USER_CONTEXT_STORAGE_FETCH_SUCCESS', () => {
      const newEntries = [{ fileID: 300 }, { fileID: 400 }];

      it('replaces the entries with new ones when `append` is false, and update list metadata', () => {
        const action = {
          type: AT.USER_CONTEXT_STORAGE_FETCH_SUCCESS,
          append: false,
          entries: newEntries,
          nextPageToken: 'YYZ',
          elementsOrder: ['fileID'],
        };
        const state = reducer(populatedState, action);
        expect(state).toMatchSnapshot();
      });

      it('combines both entry lists when `append` is true, and update list metadata', () => {
        const action = {
          type: AT.USER_CONTEXT_STORAGE_FETCH_SUCCESS,
          append: true,
          q: 'search',
          entries: newEntries,
          nextPageToken: 'YYZ',
          elementsOrder: ['fileID'],
        };
        const state = reducer(populatedState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('USER_CONTEXT_STORAGE_LIST_ITEM_ONCLICK', () => {
      it('sets the selected item', () => {
        const action = {
          type: AT.USER_CONTEXT_STORAGE_LIST_ITEM_ONCLICK,
          file: {
            fileID: 2,
          },
        };
        const state = reducer(populatedState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('USER_CONTEXT_STORAGE_FILES_UPLOAD_OPEN_MODAL', () => {
      it('sets modal visibility to `true`', () => {
        const action = {
          type: AT.USER_CONTEXT_STORAGE_FILES_UPLOAD_OPEN_MODAL,
        };
        const state = reducer(populatedState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('USER_CONTEXT_STORAGE_FILES_UPLOAD_CLOSE_MODAL', () => {
      it('sets modal visibility to `false`', () => {
        const closedState = { ...populatedState, uploadModalVisible: true };
        const action = {
          type: AT.USER_CONTEXT_STORAGE_FILES_UPLOAD_CLOSE_MODAL,
        };
        expect(closedState).toMatchSnapshot();
        const state = reducer(closedState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('USER_CONTEXT_STORAGE_FILES_UPLOAD_SUCCESS', () => {
      it('sets selectedFile and hide the modal', () => {
        const action = {
          type: AT.USER_CONTEXT_STORAGE_FILES_UPLOAD_SUCCESS,
          selectedFile: {
            fileID: 500,
          },
        };
        const state = reducer(populatedState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('USER_CONTEXT_STORAGE_DETAILS_DELETE_FILE_SUCCESS', () => {
      it('Removes the file with the given id from the list and unselect it', () => {
        const action = {
          type: USER_CONTEXT_STORAGE_DETAILS_DELETE_FILE_SUCCESS,
          fileID: 1,
        };
        const state = reducer(populatedState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('USER_CONTEXT_STORAGE_GET_STORAGE_CONTEXTS_SUCCESS', () => {
      it('sets the contexts list', () => {
        const action = {
          type: AT.USER_CONTEXT_STORAGE_GET_STORAGE_CONTEXTS_SUCCESS,
          contextsList: ['ps3', 'ps4'],
        };
        const state = reducer(populatedState, action);
        expect(state).toMatchSnapshot();
      });
    });
  });
});
