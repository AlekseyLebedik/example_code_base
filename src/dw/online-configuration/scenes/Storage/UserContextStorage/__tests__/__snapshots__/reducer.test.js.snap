// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserContextStorage Reducer USER_CONTEXT_STORAGE_DETAILS_DELETE_FILE_SUCCESS Removes the file with the given id from the list and unselect it 1`] = `
Object {
  "contextsList": Array [
    "x360",
    "x1",
  ],
  "elementsOrder": Array [
    "fileID",
  ],
  "entries": Array [
    Object {
      "fileID": 2,
    },
  ],
  "nextPageToken": "ABC",
  "q": undefined,
  "reloadDetailPath": "/a",
  "selectedFile": undefined,
  "uploadModalVisible": false,
}
`;

exports[`UserContextStorage Reducer USER_CONTEXT_STORAGE_FETCH_SUCCESS combines both entry lists when \`append\` is true, and update list metadata 1`] = `
Object {
  "contextsList": Array [
    "x360",
    "x1",
  ],
  "elementsOrder": Array [
    "fileID",
  ],
  "entries": Array [
    Object {
      "fileID": 1,
    },
    Object {
      "fileID": 2,
    },
    Object {
      "fileID": 300,
    },
    Object {
      "fileID": 400,
    },
  ],
  "nextPageToken": "ABC",
  "q": "search",
  "reloadDetailPath": "/a",
  "selectedFile": undefined,
  "uploadModalVisible": false,
}
`;

exports[`UserContextStorage Reducer USER_CONTEXT_STORAGE_FETCH_SUCCESS replaces the entries with new ones when \`append\` is false, and update list metadata 1`] = `
Object {
  "contextsList": Array [
    "x360",
    "x1",
  ],
  "elementsOrder": Array [
    "fileID",
  ],
  "entries": Array [
    Object {
      "fileID": 300,
    },
    Object {
      "fileID": 400,
    },
  ],
  "nextPageToken": "ABC",
  "q": undefined,
  "reloadDetailPath": "/a",
  "selectedFile": undefined,
  "uploadModalVisible": false,
}
`;

exports[`UserContextStorage Reducer USER_CONTEXT_STORAGE_FILES_UPLOAD_CLOSE_MODAL sets modal visibility to \`false\` 1`] = `
Object {
  "contextsList": Array [
    "x360",
    "x1",
  ],
  "elementsOrder": Array [
    "fileID",
  ],
  "entries": Array [
    Object {
      "fileID": 1,
    },
    Object {
      "fileID": 2,
    },
  ],
  "nextPageToken": "ABC",
  "q": undefined,
  "reloadDetailPath": "/a",
  "selectedFile": Object {
    "fileID": 1,
  },
  "uploadModalVisible": true,
}
`;

exports[`UserContextStorage Reducer USER_CONTEXT_STORAGE_FILES_UPLOAD_CLOSE_MODAL sets modal visibility to \`false\` 2`] = `
Object {
  "contextsList": Array [
    "x360",
    "x1",
  ],
  "elementsOrder": Array [
    "fileID",
  ],
  "entries": Array [
    Object {
      "fileID": 1,
    },
    Object {
      "fileID": 2,
    },
  ],
  "nextPageToken": "ABC",
  "q": undefined,
  "reloadDetailPath": "/a",
  "selectedFile": Object {
    "fileID": 1,
  },
  "uploadModalVisible": false,
}
`;

exports[`UserContextStorage Reducer USER_CONTEXT_STORAGE_FILES_UPLOAD_OPEN_MODAL sets modal visibility to \`true\` 1`] = `
Object {
  "contextsList": Array [
    "x360",
    "x1",
  ],
  "elementsOrder": Array [
    "fileID",
  ],
  "entries": Array [
    Object {
      "fileID": 1,
    },
    Object {
      "fileID": 2,
    },
  ],
  "nextPageToken": "ABC",
  "q": undefined,
  "reloadDetailPath": "/a",
  "selectedFile": Object {
    "fileID": 1,
  },
  "uploadModalVisible": true,
}
`;

exports[`UserContextStorage Reducer USER_CONTEXT_STORAGE_FILES_UPLOAD_SUCCESS sets selectedFile and hide the modal 1`] = `
Object {
  "contextsList": Array [
    "x360",
    "x1",
  ],
  "elementsOrder": Array [
    "fileID",
  ],
  "entries": Array [
    Object {
      "fileID": 1,
    },
    Object {
      "fileID": 2,
    },
  ],
  "nextPageToken": "ABC",
  "q": undefined,
  "reloadDetailPath": "/a",
  "selectedFile": Object {
    "fileID": 500,
  },
  "uploadModalVisible": false,
}
`;

exports[`UserContextStorage Reducer USER_CONTEXT_STORAGE_GET_STORAGE_CONTEXTS_SUCCESS sets the contexts list 1`] = `
Object {
  "contextsList": Array [
    "ps3",
    "ps4",
  ],
  "elementsOrder": Array [
    "fileID",
  ],
  "entries": Array [
    Object {
      "fileID": 1,
    },
    Object {
      "fileID": 2,
    },
  ],
  "nextPageToken": "ABC",
  "q": undefined,
  "reloadDetailPath": "/a",
  "selectedFile": Object {
    "fileID": 1,
  },
  "uploadModalVisible": false,
}
`;

exports[`UserContextStorage Reducer USER_CONTEXT_STORAGE_LIST_ITEM_ONCLICK sets the selected item 1`] = `
Object {
  "contextsList": Array [
    "x360",
    "x1",
  ],
  "elementsOrder": Array [
    "fileID",
  ],
  "entries": Array [
    Object {
      "fileID": 1,
    },
    Object {
      "fileID": 2,
    },
  ],
  "nextPageToken": "ABC",
  "q": undefined,
  "reloadDetailPath": "/a",
  "selectedFile": Object {
    "fileID": 2,
  },
  "uploadModalVisible": false,
}
`;

exports[`UserContextStorage Reducer USER_CONTEXT_STORAGE_RELOAD_DETAILS sets the reload details path 1`] = `
Object {
  "contextsList": Array [
    "x360",
    "x1",
  ],
  "elementsOrder": Array [
    "fileID",
  ],
  "entries": Array [
    Object {
      "fileID": 1,
    },
    Object {
      "fileID": 2,
    },
  ],
  "nextPageToken": "ABC",
  "q": undefined,
  "reloadDetailPath": "/b",
  "selectedFile": Object {
    "fileID": 1,
  },
  "uploadModalVisible": false,
}
`;

exports[`UserContextStorage Reducer returns default state 1`] = `
Object {
  "contextsList": Array [],
  "elementsOrder": Array [],
  "entries": Array [],
  "q": undefined,
  "reloadDetailPath": null,
  "selectedFile": undefined,
  "uploadModalVisible": false,
}
`;
